core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadOnlyLDAPUserStoreManager.java

```diff
@@ -2917,23 +2923,40 @@ private void getExpressionConditionsAsList(Condition condition, List<ExpressionC
         return users;
     }
 
+    /**
+     * Method to verify whether the filter operation is a single attribute filter scenario or multi attribute filter scenario
+     *
+     * @param sizeOfExpressionConditions How many sub parts does Expression Condition has (value will be equals to 1
+     *                                   for a single attribute filter)
+     * @return True if the operation is a single attribute filter.
+     */
+    private boolean verifyFilterOperation(int sizeOfExpressionConditions) {
```

Comment:
As this returns true only if it's a single attribute filter better to denote that in method name

Reply:
Fixed: 9f50ebec3b09439da2c0332c6abd48f49d45ff17